### PR TITLE
Support properties in plugins

### PR DESCRIPTION
**Pull request by mik-laj**
---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Target Github ISSUE in description if exists
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/plugins_manager.py [hunk 1]
```diff
@@ -145,11 +145,12 @@ def load_entrypoint_plugins():
     for entry_point in entry_points:  # pylint: disable=too-many-nested-blocks
         log.debug('Importing entry_point plugin %s', entry_point.name)
         try:
-            plugin_obj = entry_point.load()
-            if is_valid_plugin(plugin_obj):
+            plugin_class = entry_point.load()
+            if is_valid_plugin(plugin_class):
+                plugin_obj = plugin_class()
```
Comment (turbaszek) [inline airflow/plugins_manager.py]:
```suggestion
                plugin_instance = plugin_class()
```
WDYT?

File: airflow/plugins_manager.py [hunk 2]
```diff
@@ -145,11 +145,12 @@ def load_entrypoint_plugins():
...
                 if callable(getattr(plugin_obj, 'on_load', None)):
                     plugin_obj.on_load()
-                    plugins.append(plugin_obj)
+                    plugins.append(plugin_obj())
```
Comment (turbaszek) [inline airflow/plugins_manager.py]: I'm not sure, but do we call here an instantiated class (via `__call__`)? Why we do this?

Comment (mik-laj) [reply airflow/plugins_manager.py]: We call the class constructor.Previously, we only used the class, not the plugin object.
```
>>> class AirflowTestPropertyPlugin(AirflowPlugin):
...     name = "test_property_plugin"
...     @property
...     def operators(self):
...         from airflow.models.baseoperator import BaseOperator
...         class PluginPropertyOperator(BaseOperator):
...             pass
...         return [PluginPropertyOperator]
...
>>> print(AirflowTestPropertyPlugin().operators)
[<class '__main__.AirflowTestPropertyPlugin.operators.<locals>.PluginPropertyOperator'>]
>>> print(AirflowTestPropertyPlugin.operators)
<property object at 0x10b2172c8>

```

File: airflow/plugins_manager.py (hunk 1 above)
Comment (mik-laj) [reply airflow/plugins_manager.py]: I updated the PR.

File: airflow/plugins_manager.py (hunk 2 above)
Comment (turbaszek) [reply airflow/plugins_manager.py]: Currently we have:
```python
plugin_class = entry_point.load() # class object
plugin_obj = plugin_class() # class instance created via calling __init__
plugin_obj() # we call existing instance so it has to have __call__
```
What am I missing?

Comment (mik-laj) [reply airflow/plugins_manager.py]: Good point. Fixed.

**Review by turbaszek — APPROVED** [review]

**Pull request merged**